compiler/ballerina-lang/src/main/resources/grammar/BallerinaLexer.g4

```diff
@@ -976,20 +988,21 @@ StringTemplateText
 
 fragment
 StringTemplateStringChar
-    :   ~[`{\\]
-    |   '\\' [`{]
+    :   ~[`${\\]
     |   WS
     |   StringLiteralEscapedSequence
     ;
 
 fragment
 StringLiteralEscapedSequence
     :   '\\\\'
-    |   '\\{{'
+    |   '\\' ~'$'
```

Comment:
shall we look into this bit more? like this will allow below string which IMO is invalid

string ` sss\sdfd\a ${kk} df`

Reply:
Agreed! Updated the PR to disallow the above and similar cases. Along with this change, I have also simplified the string template literal rule as we discussed offline about the possibilities of doing so.